web/packages/teleterm/src/services/grpcCredentials/files.ts

```diff
@@ -64,36 +67,40 @@ export async function readGrpcCert(
     return !!(await stat(fullPath)).size;
   }
 
-  function watchForFile(): Promise<Buffer> {
+  function waitForFile(): Promise<Buffer> {
     return new Promise((resolve, reject) => {
-      abortController.signal.onabort = () => {
-        watcher.close();
-        clearTimeout(timeout);
-      };
-
-      const timeout = setTimeout(() => {
-        reject(
-          `Could not read ${certName} certificate. The operation timed out.`
-        );
-      }, 10_000);
+      wait(timeoutMs, abortController.signal).then(
+        () =>
+          reject(
+            new Error(
+              `Could not read ${certName} certificate within the timeout.`
+            )
+          ),
+        () => {} // Ignore abort errors.
```

Comment:
I don't get it why I need to manually ignore errors here and below when I call `fileExistsAndHasSize` on line 95. If I don't do it here then the test fails with an abort error. If I don't do it below, then the test fails with a stat error.

Is this a custom behavior of Jest that's supposed to make sure you don't have unhandled promise errors? Because I don't see how the state of the promise returned by `wait` or `fileExistsAndHasSize` should have any effect on the state of the promise we're returning from `waitForFile`. Errors thrown in the promise executor cause the promise to be rejected with that error, but in this case we're not throwing any errors, they're all contained within promises.